### PR TITLE
Make varlink API more idiomatic

### PR DESCRIPTION
**Pull request by larskarlitski**
See individual commits. This is a first round of fixes I've found when I started working with this API for [cockpit-podman](https://github.com/cockpit-project/cockpit-podman). I have some more, but these were the ones I could easily implement.

I hope that you don't consider the varlink API to be stable, yet? As far as I know, cockpit-podman is the only user so far and I'm fine updating that on the next release.

### CODE REVIEW

**Comment by larskarlitski** [comment]
Hm, I don't understand the latest error on the "gating" check. It says "9 commits to fix", but not what the fixes should be...

File: cmd/podman/varlink/io.podman.varlink [hunk 1]
```diff
@@ -393,34 +372,29 @@ type Runlabel(
...
-method ListContainers() -> (containers: []ListContainerData)
+# ListContainers returns information about all containers.
+# See also [GetContainer](#GetContainer).
+method ListContainers() -> (containers: []Container)
 
-# GetContainer takes a name or ID of a container and returns single ListContainerData
-# structure.  A [ContainerNotFound](#ContainerNotFound) error will be returned if the container cannot be found.
-# See also [ListContainers](ListContainers) and [InspectContainer](#InspectContainer).
-method GetContainer(name: string) -> (container: ListContainerData)
+# Returns information about a single container.  If a container with the given
```
Comment (mheon) [inline cmd/podman/varlink/io.podman.varlink]: I think we'd prefer that all descriptions start with the name of the API endpoint, for consistency (so `GetContainer returns...` instead)

File: pkg/varlinkapi/system.go [hunk 2]
```diff
@@ -15,22 +16,14 @@ func (i *LibpodAPI) GetVersion(call iopodman.VarlinkCall) error {
...
-		Go_version:         versionInfo.GoVersion,
-		Git_commit:         versionInfo.GitCommit,
-		Built:              versionInfo.Built,
-		Os_arch:            versionInfo.OsArch,
-	})
-}
-
-// Ping returns a simple string "OK" response for clients to make sure
-// the service is working.
-func (i *LibpodAPI) Ping(call iopodman.VarlinkCall) error {
```
Comment (mheon) [inline pkg/varlinkapi/system.go]: @baude Any reason to keep this? I think it might be useful for a liveliness check

**Comment by mheon** [comment]
Gitvalidation is complaining that you need a signoff on your commits

@baude Will the time format changes here mean we need changes in `adapter` to read times back in as RFC3339?

**Comment by mheon** [comment]
/ok-to-test
bot, add author to whitelist

**Comment by baude** [comment]
@larskarlitski cockpit is not the only user of the API ... the remote client for darwin I am writing is the a major one.  Unfortunately we don't quite have integration tests running for those yet as we are still working on a couple of container related ones.  Until that is resolved, the tests cannot be run.  So you will have to make sure you don't break that either.

**Comment by baude** [comment]
please dont remove endpoints that are not implemented.  it is the only list we have for what needs to be done still.

File: pkg/varlinkapi/system.go (hunk 2 above)
Comment (baude) [reply pkg/varlinkapi/system.go]: yes, do not remove it; we often use it to test that the connection is working.

File: cmd/podman/varlink/io.podman.varlink [hunk 3]
```diff
@@ -8,7 +8,7 @@ type Version (
   version: string,
   go_version: string,
   git_commit: string,
-  built: int,
+  built: string, # as RFC3339
```
Comment (baude) [inline cmd/podman/varlink/io.podman.varlink]: can you verify this still works with the remote client?

File: cmd/podman/varlink/io.podman.varlink [hunk 4]
```diff
@@ -386,9 +376,15 @@ type Runlabel(
     opts: [string]string
 )
 
-# GetVersion returns a Version structure describing the libpod setup on their
-# system.
-method GetVersion() -> (version: Version)
```
Comment (baude) [inline cmd/podman/varlink/io.podman.varlink]: please add API documentation like many of the other functions. thanks.

**Comment by baude** [comment]
@larskarlitski great start ... i like the changes for images and containers naming. if you can address my comments, that would be great.

**Comment by rhatdan** [comment]
@larskarlitski Have you had a chance to address @baude Issues?

File: cmd/podman/varlink/io.podman.varlink (hunk 1 above)
Comment (larskarlitski) [reply cmd/podman/varlink/io.podman.varlink]: Ooh I hadn't noticed that, but I like it. Changed the ones I touched to be consistent again.

File: pkg/varlinkapi/system.go (hunk 2 above)
Comment (larskarlitski) [reply pkg/varlinkapi/system.go]: Hm, I don't feel super strong about it, but it seems weird to me to have a debug call like this in the API. Could you instead use `GetVersion()` for testing that the connection is working?

File: cmd/podman/varlink/io.podman.varlink (hunk 3 above)
Comment (larskarlitski) [reply cmd/podman/varlink/io.podman.varlink]: Sure. Where can I find this client?

File: cmd/podman/varlink/io.podman.varlink (hunk 4 above)
Comment (larskarlitski) [reply cmd/podman/varlink/io.podman.varlink]: Added it back.

**Comment by larskarlitski** [comment]
> Gitvalidation is complaining that you need a signoff on your commits

Thanks. Added those and addressed @baude's comments. Thanks for the reviews!

> please dont remove endpoints that are not implemented. it is the only list we have for what needs to be done still.

We can't commit to a stable API if some of the functions return a `NotImplemented` type. Better to leave out the things that aren't there yet. to track missing functions, would you be ok with me adding a list in a `TODO` file or similar?

**Comment by baude** [comment]
> > Gitvalidation is complaining that you need a signoff on your commits
> 
> Thanks. Added those and addressed @baude's comments. Thanks for the reviews!
> 
> > please dont remove endpoints that are not implemented. it is the only list we have for what needs to be done still.
> 
> We can't commit to a stable API if some of the functions return a `NotImplemented` type. Better to leave out the things that aren't there yet. to track missing functions, would you be ok with me adding a list in a `TODO` file or similar?

there are several i dont mind pruning.  Ill remove those in a different PR.  Can we compromise and instead of removing, comment them out?

**Comment by larskarlitski** [comment]
> there are several i dont mind pruning. Ill remove those in a different PR. Can we compromise and instead of removing, comment them out?

Good idea! Done.

Are these test failures related to my changes? I can't see how they would be...

**Comment by mheon** [comment]
Restarted one of the cirrus tests to see if this is just flakes, or we have a real issue

@baude Mind taking another look at this?

**Comment by baude** [comment]
kind of baffled by this one; the tests look like their legitimately failing?

**Comment by larskarlitski** [comment]
> kind of baffled by this one; the tests look like their legitimately failing?

I don't understand all the tests and it's hard to get them working locally. I've found one place where the remoteClient panics in `splitStringDate`, because it still assumed the old time format. I could remove this function, because we're now just passing along standard time stamps.

**Comment by rhatdan** [comment]
@baude @jwhonce Tests are passing PTAL

**Comment by baude** [comment]
please rebase and I will merge it @larskarlitski

**Comment by larskarlitski** [comment]
Rebased.

**Comment by larskarlitski** [comment]
Rebased, again.

**Comment by baude** [comment]
@larskarlitski I will watch your PR .. .btw, join us on #podman on freenode!

**Pull request merged**